src/api/app/controllers/webui/session_controller.rb

```diff
@@ -8,35 +8,30 @@ def new
   end
 
   def create
-    user = User.find_with_credentials(params[:username], params[:password])
+    session_creator = SessionControllerService::SessionCreator.new(params.slice(:username, :password)).call
 
-    unless user
-      RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui,failure=unauthenticated value=1')
+    unless session_creator.valid? && session_creator.exist?
+      SessionControllerNotificationJob.perform_later('login,access_point=webui,failure=unauthenticated value=1')
       redirect_to(new_session_path, error: 'Authentication failed')
       return
     end
 
-    unless user.is_active?
-      RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui,failure=disabled value=1')
+    unless session_creator.user.is_active?
+      SessionControllerNotificationJob.perform_later('login,access_point=webui,failure=disabled value=1')
       redirect_to(root_path, error: 'Your account is disabled. Please contact the administrator for details.')
       return
     end
 
-    User.session = user
-    session[:login] = user.login
-    Rails.logger.debug "Authenticated as user '#{user.login}'"
-    RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui value=1')
-
+    User.session = session_creator.user
+    session[:login] = session_creator.user.login
+    SessionControllerNotificationJob.perform_later('login,access_point=webui value=1')
     redirect_on_login
   end
 
   def destroy
-    Rails.logger.info "Logging out: #{session[:login]}"
-
     reset_session
-    RabbitmqBus.send_to_bus('metrics', 'logout,access_point=webui value=1')
+    SessionControllerNotificationJob.perform_later('logout,access_point=webui value=1')
```

Comment:
This doesn't make much sense, pushing to the bus is not costly. And the resulting data only makes sense if it's connected to the time where it happened, not the time where DJ has gone through it's queue.

Reply:
it's updating a counter.. just it, right? I'm not sure why the time should matter.. even if it's not "costly" the session creation shouldn't be affected by that. Normal applications always off-load main thread with logging actions.

Reply:
> I'm not sure why the time should matter..

Because this is sent to a timeseries database. You correlate it with other data from this DB, logs, etc. Timestamps matter very much.

> IMO it's a good pattern to off-load pure logging actions from main thread..

I don't agree. You wouldn't do this with `Rails.logger` would you?

Reply:
> Because this is sent to a timeseries database. You correlate it with other data from this DB, logs, etc. Timestamps matter very much.

We are sending it to a Remote server and we are aggregating the "+1", so I don't see how we are able to correlate it with something else.

> I don't agree. You wouldn't do this with Rails.logger would you?

if we are just counting events why not? `Thread.new { Rails.logger("+1") }

Would I make you happy if I pass the timestamp around?